yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -483,14 +483,14 @@ private[spark] class ApplicationMaster(
         } catch {
           case e: InvocationTargetException =>
             e.getCause match {
-              case _: InterruptedException =>
+              case e: InterruptedException =>
                 // Reporter thread can interrupt to stop user class
-              case e: Exception =>
+              case _ =>
```

Comment:
So, the problem is that this was not catching `Error`, but just `Exception`, right?

I think this generates a compiler warning, so probably should be `case _: Throwable =>`. And you can avoid the duplicate `getCause` calls by giving this variable a name instead of using the placeholder.